packages/react/src/components/TextArea/index.tsx

```diff
@@ -126,43 +91,53 @@ const TextArea = forwardRef<HTMLTextAreaElement, TextAreaProps>(
 
     const containerRef = useRef(null)
 
-    useFocusWithClick(containerRef, ariaRef)
+    useFocusWithClick(containerRef, textAreaRef)
 
-    const inputProps = mergeProps(restProps, ariaInputProps)
+    const textAreaId = useId(props.id)
+    const describedbyId = useId()
+    const labelledbyId = useId()
+
+    const showAssistiveText =
+      assistiveText != null && assistiveText.length !== 0
 
     return (
       <TextFieldRoot className={className} isDisabled={disabled}>
         <TextFieldLabel
+          htmlFor={textAreaId}
+          id={labelledbyId}
           label={label}
-          requiredText={requiredText}
           required={required}
+          requiredText={requiredText}
           subLabel={subLabel}
-          {...labelProps}
           {...(!showLabel ? visuallyHiddenProps : {})}
         />
         <StyledTextareaContainer
+          aria-disabled={disabled === true ? 'true' : undefined}
+          invalid={props.invalid === true}
           ref={containerRef}
-          invalid={invalid}
           rows={showCount ? rows + 1 : rows}
-          aria-disabled={disabled === true ? 'true' : undefined}
         >
           <StyledTextarea
-            ref={mergeRefs(textareaRef, forwardRef, ariaRef)}
-            rows={rows}
+            aria-describedby={describedbyId}
```

Comment:
I noticed that aria-describedby is set up, but there are instances where the DOM element with the specified ID gets removed. It might be worth considering removing aria-describedby depending on the presence or absence of showAssistiveText to ensure accessibility features behave as expected and do not reference non-existent DOM elements. 

https://developer.mozilla.org/ja/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#:~:text=%E5%90%8C%E3%81%98%20DOM%20%E6%96%87%E6%9B%B8%E5%86%85%E3%81%AE%E8%A6%81%E7%B4%A0%E3%82%92%E5%8F%82%E7%85%A7%E3%81%97%E3%81%AA%E3%81%91%E3%82%8C%E3%81%B0%E3%81%AA%E3%82%8A%E3%81%BE%E3%81%9B%E3%82%93%E3%80%82

Reply:
You are right. I've fixed the issue.